### PR TITLE
BUG: disallow scalar in CategoricalIndex construtor

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
i guess need to deprecate here as well (unless you use the Categorical for constrution)

**Comment by jbrockmendel** [comment]
> i guess need to deprecate here as well

i think this is pretty unambiguously a bug

**Comment by jreback** [comment]
> > i guess need to deprecate here as well
> 
> i think this is pretty unambiguously a bug

oh right yeah its a scalar in an Index, ok

**Comment by jreback** [comment]
ok i guess add a whatsnew here to document

**Comment by jbrockmendel** [comment]
whatsnew added + greenish

File: pandas/core/indexes/category.py [hunk 1]
```diff
@@ -186,11 +186,7 @@ def __new__(
         name = maybe_extract_name(name, data, cls)
 
         if is_scalar(data):
-            # don't allow scalars
-            # if data is None, then categories must be provided
-            if data is not None or categories is None:
-                raise cls._scalar_data_error(data)
-            data = []
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/category.py:193]: This is not actually about passing scalar data, but about passing no data (eg `pd.CategoricalIndex(categories=['a', 'b'])`). We generally support constructors with passing no data to create an empty (eg Series also allows this). 

So I would either keep supporting this, or either it should at least be deprecated (this eg breaks dask)

Comment (jorisvandenbossche) [reply pandas/core/indexes/category.py:193]: Opened https://github.com/pandas-dev/pandas/issues/38944 to track this

**Pull request merged**